### PR TITLE
Migrate to Dokka 0.10.1

### PR DESCRIPTION
**Pull request by fbenz**
Changes to the Dokka extension:
* Upgrade to Dokka 0.10.1 (see [0.10.1](https://github.com/Kotlin/dokka/releases/tag/0.10.1) and [0.10.0](https://github.com/Kotlin/dokka/releases/tag/0.10.0))
* Build a test JAR instead of copying the code over. There have been quite some changes in in the test helpers lately.
* Add an integration test to catch potential setup and dependency injection issues. The fix for the DI issue I ran into is the same as in Dokka itself: https://github.com/Kotlin/dokka/blob/de2f32d91fb6f564826ddd7644940452356e2080/core/src/main/kotlin/Formats/FormatDescriptor.kt#L28
* I made sure in the POM file that it has to be build with Java 8.

Example project:
* There is no need for Java 11 dependencies or the Javadoc Doclet in the Gradle build file of the example project.
* `reportUndocumented = true` is the annoying default and thus I changed it to false.
* I have the checked the resulting documentation and there is no difference.

### CODE REVIEW

File: spring-auto-restdocs-docs/contributing.adoc [hunk 1]
```diff
@@ -57,6 +72,49 @@ Afterwards the test JAR is located at
...
+
+[source,groovy]
+----
+task testJar(type: Jar) {
+    classifier = 'tests'
+    from sourceSets.test.output
+}
+----
+
+To speed the build up and avoid test issues, once can reduce the list ob subprojects in
```
Comment (jmisur) [inline spring-auto-restdocs-docs/contributing.adoc:103]: of

File: spring-auto-restdocs-docs/contributing.adoc [hunk 2]
```diff
@@ -57,6 +72,49 @@ Afterwards the test JAR is located at
...
+
+.Bash
+[source,bash]
+----
+git clone git@github.com:Kotlin/dokka.git
+cd dokka
+git checkout ${dokka.version}
+----
+
+Add the following snippted at the end of `core/build.gradle`:
```
Comment (jmisur) [inline spring-auto-restdocs-docs/contributing.adoc:93]: snippet

File: spring-auto-restdocs-dokka-json/src/test/kotlin/capital/scalable/dokka/json/JsonFormatIntegrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    override val moduleName: String = "",
+    override val sourceRoots: List<DokkaConfiguration.SourceRoot> = emptyList(),
+    override val samples: List<String> = emptyList(),
+    override val includes: List<String> = emptyList(),
+    override val includeNonPublic: Boolean = false,
+    override val includeRootPackage: Boolean = false,
+    override val reportUndocumented: Boolean = false,
+    override val skipEmptyPackages: Boolean = false,
+    override val skipDeprecated: Boolean = false,
+    override val jdkVersion: Int = 6,
```
Comment (jmisur) [inline spring-auto-restdocs-dokka-json/src/test/kotlin/capital/scalable/dokka/json/JsonFormatIntegrationTest.kt:104]: 6?

File: spring-auto-restdocs-docs/contributing.adoc [hunk 4]
```diff
@@ -19,7 +19,22 @@ mvn install:install-file -Dfile=lib/spring-restdocs-core-${spring-restdocs.versi
...
+
+Dokka publishes no test JAR, but the Spring Auto REST Docs Dokka extension uses their test utilities.
+If you want to build this project yourself, you need to install the test JAR.
+The test JAR is located in the lib folder and can be installed as follow:
+
+.Bash
+[source,bash]
+----
+mvn install:install-file -Dfile=lib/dokka-core-${dokka.version}-tests.jar
+     -DgroupId=org.jetbrains.dokka -DartifactId=dokka-core -Dversion=${dokka.version}
```
Comment (jmisur) [inline spring-auto-restdocs-docs/contributing.adoc:33]: missing `\` for multiline command

File: spring-auto-restdocs-dokka-json/src/test/kotlin/capital/scalable/dokka/json/JsonFormatIntegrationTest.kt (hunk 3 above)
Comment (fbenz) [reply spring-auto-restdocs-dokka-json/src/test/kotlin/capital/scalable/dokka/json/JsonFormatIntegrationTest.kt]: It does not matter for the test but I changed it to 8 to have it in line with what we actually use.

File: spring-auto-restdocs-docs/contributing.adoc (hunk 1 above)
Comment (fbenz) [reply spring-auto-restdocs-docs/contributing.adoc]: Thanks, by using a spell checker I found more :)

**Pull request merged**